### PR TITLE
Added Documentation for Spider.update_settings

### PR DESCRIPTION
**Pull request by tuhinmallick**
I added documentation for Spider.update_settings in relation to the issue: #5745 
My contribution included a detailed explanation and a code example.

### CODE REVIEW

File: docs/topics/spiders.rst [hunk 1]
```diff
@@ -261,6 +261,32 @@ to give data more structure you can use :class:`~scrapy.Item` objects::
 
 .. _spiderargs:
 
+    .. method:: update_settings(settings)
+
+        This is a classmethod which is called to override the configuration of your spider.
+        It receives the settings and passes :attr:`custom_settings` or `{}` to ``setdict`` 
+        method to register the setting with priority ``spider``. 
+        It is preferrable to use the :attr:`custom_settings` when making spiders class
```
[the remark this answers is not in the document]
Comment (tuhinmallick) [reply docs/topics/spiders.rst:269]: Ah typos. Done

File: docs/topics/spiders.rst [hunk 2]
```diff
@@ -261,6 +261,32 @@ to give data more structure you can use :class:`~scrapy.Item` objects::
 
 .. _spiderargs:
```
Comment (wRAR) [inline docs/topics/spiders.rst:262]: This label belongs to the `Spider arguments` section below.

File: docs/topics/spiders.rst [hunk 3]
```diff
@@ -261,6 +261,32 @@ to give data more structure you can use :class:`~scrapy.Item` objects::
...
+    import scrapy
+
+
+    class MySpider(scrapy.Spider):
+        name = 'example.com'
+        allowed_domains = ['example.com']
+        start_urls = ['http://www.example.com/feed.csv']
+        spider_settings = {'TEST1': 'spider', 'TEST2': 'spider'}
+        project_settings = {'TEST1': 'project', 'TEST3': 'project'}
+        self.spider_class.custom_settings = spider_settings
```
Comment (wRAR) [inline docs/topics/spiders.rst:285]: This code won't work.

Comment (tuhinmallick) [reply docs/topics/spiders.rst:283]: Do I need to provide a code example ?

Comment (wRAR) [reply docs/topics/spiders.rst:283]: You've already provided a code example, but it's incorrect.

Comment (tuhinmallick) [reply docs/topics/spiders.rst]: @wRAR Fixed the example. Please have a look into it.

Comment (wRAR) [reply docs/topics/spiders.rst]: Sorry but this code makes no sense. It also doesn't even use `update_settings()`.

Comment (tuhinmallick) [reply docs/topics/spiders.rst]: I tried to show an example of how to update the settings ( I included update_settings()). if the example is not needed, I can remove the example altogether.

Comment (wRAR) [reply docs/topics/spiders.rst]: This is not even an example of defining the `update_settings()` method.

Comment (tuhinmallick) [reply docs/topics/spiders.rst]: I  got this example from the unit tests. So, I guess the unit tests are wrong and needs to be configured too.

Comment (wRAR) [reply docs/topics/spiders.rst]: Sorry but that's the wrong place to get an example (I would even say that if you don't know how to use this method you shouldn't write docs and examples for it).

File: docs/topics/spiders.rst [hunk 4]
```diff
@@ -259,6 +259,35 @@ to give data more structure you can use :class:`~scrapy.Item` objects::
...
+    class MySpider(scrapy.Spider):
+        spider_class = Spider
+        name = 'example.com'
+        allowed_domains = ['example.com']
+        start_urls = ['http://www.example.com/feed.csv']
+        spider_settings = {'TEST1': 'spider', 'TEST2': 'spider'}
+        project_settings = {'TEST1': 'project', 'TEST3': 'project'}
+        spider_class.custom_settings = spider_settings
+        settings = Settings(project_settings, priority='project')
+        spider_class.update_settings(settings)
```
Comment (Gallaecio) [inline docs/topics/spiders.rst:288]: This is the wrong place and indentation (it should be after “This is a classmethod…”, with the same indentation for “Let’s see an example”), and the example is wrong (see https://github.com/scrapy/scrapy/pull/5757 for a better example, yet at the moment also incorrect).

**Pull request closed without merging**